src/test/unit/ilp.test.js

```diff
@@ -36,35 +39,39 @@ describe('ILP', () => {
         expect(condition).toBeTruthy();
     });
 
-
-    test('ILP packet should contain a valid transaction object', () => {
-        const {ilpPacket} = ilp.getQuoteResponseIlp(quoteRequest, partialResponse);
-
+    test('deserializes the ILP packet into a valid transaction object', () => {
+        // Arrange
+        const { ilpPacket } = ilp.getQuoteResponseIlp(quoteRequest, partialResponse);
+        const expectedDataElement = {
+            // Not all elements from quoteRequest end up in the ilp packet
+            amount: {
+                amount: '500',
+                currency: 'USD'
+            },
+            payee: quoteRequest.payee,
+            payer: quoteRequest.payer,
+            transactionType: quoteRequest.transactionType,
+            quoteId: quoteRequest.quoteId,
+            transactionId: quoteRequest.transactionId,
+        };
+
+        // Act
         const binaryPacket = Buffer.from(ilpPacket, 'base64');
         const jsonPacket = IlpPacket.deserializeIlpPacket(binaryPacket);
-        console.log(`Decoded ILP packet: ${util.inspect(jsonPacket)}`);
-
         const dataElement = JSON.parse(Buffer.from(jsonPacket.data.data.toString('utf8'), 'base64').toString('utf8'));
 
-        console.log(`Decoded ILP packet data element: ${util.inspect(dataElement)}`);
+        // Assert
+        expect(dataElement).toStrictEqual(expectedDataElement);
     });
 
-
     test('ILP fulfilment should match condition', () => {
-        const {fulfilment, ilpPacket, condition} = ilp.getQuoteResponseIlp(quoteRequest, partialResponse);
```

Comment:
@lewisdaly I think you are trying to simplify this code by using the Ilp class to check itself. A good intention but, this was deliberately avoided to prevent cases where a broken implementation may incorrectly validate itself. I use the "original source of truth" i.e. the IlpPacket library to check the encoding/decoding to avoid that.

Reply:
Thanks @bushjames. I think I understand. Perhaps I just need to add assertions to make sure some of the implicit tests (that were originally in comments) are appropriately handled.

Reply:
@bushjames I've updated this test to put the `ilpPacket` back, and ensure that it deserializes correctly. I think this addresses your concerns, but please let me know if there's something I'm missing.

Reply:
Thanks @lewisdaly